File: LeetCode/14_Longest_Common_Prefix/14_Longest_Common_Prefix.go
package main

import "fmt"

func longestCommonPrefix(strs []string) string {
	var result string

	for i, v := range strs[0] {
		for _, str := range strs {
			if len(str) <= i || str[i] != byte(v) {
				return result
			}
		}
		result += string(v)
	}
	return result
}

func main() {

	input := []string{"flower", "flow", "flight"}
	fmt.Print(longestCommonPrefix(input))
}

File: LeetCode/14_Longest_Common_Prefix/14_Longest_Common_Prefix.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        val result = StringBuilder()
        for ((i, c) in strs[0].withIndex()) {
            for (str in strs) {
                if (str.length <= i || str[i] != c) {
                    return result.toString()
                }
            }
            result.append(c)
        }
        return result.toString()
    }
}

fun main() {
    val s = Solution()

    val input = arrayOf("flower", "flow", "flight")
    print(s.longestCommonPrefix(input))
}
